### PR TITLE
set up image-pushing for node-ipam-controller.

### PR DESCRIPTION
**Pull request by liangyuanpeng**
/assign @aojea 
/assign @ameukam

Part of https://github.com/kubernetes-sigs/node-ipam-controller/issues/15

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        - ^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+      spec:
+        serviceAccountName: gcb-builder
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/image-builder:v20230711-e33377c2b4
+            command:
+              - /run.sh
+            args:
+              - --project=k8s-staging-node-ipam-controller
+              - --scratch-bucket=gs://k8s-staging-node-ipam-controller-gcb
```
Comment (aojea) [inline config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml:21]: have we created this bucket already?

Comment (aojea) [reply config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml:21]: see instructions https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/README.md

ping infra people for guidance, @ameukam and @upodroid

Comment (liangyuanpeng) [reply config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml:21]: Thanks, send the PR https://github.com/kubernetes/k8s.io/pull/6645 for it.

**Comment by liangyuanpeng** [comment]
@aojea  PTAL,Thanks

**Comment by liangyuanpeng** [comment]
/hold 

need to update the project bucket https://github.com/kubernetes-sigs/node-ipam-controller/blob/1323bc8a3524b800c5c757d06324a6978e81bf6d/Makefile#L18

**Comment by aojea** [comment]
> /hold
> 
> need to update the project bucket https://github.com/kubernetes-sigs/node-ipam-controller/blob/1323bc8a3524b800c5c757d06324a6978e81bf6d/Makefile#L18

/lgtm
/approve

unhold once you fix the references on the node-ipam-controller

File: config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        serviceAccountName: gcb-builder
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/image-builder:v20230711-e33377c2b4
+            command:
+              - /run.sh
+            args:
+              - --project=k8s-staging-networking
+              - --scratch-bucket=gs://k8s-staging-networking-gcb
+              - --env-passthrough=PULL_BASE_REF
+              - .
```
Comment (aojea) [inline config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml:24]:
```suggestion
              - --build-dir=.
              - --with-git-dir
              - .
```

you need to pass the git dir for the script in the makefile to get the right value

https://github.com/kubernetes-sigs/node-ipam-controller/blob/1323bc8a3524b800c5c757d06324a6978e81bf6d/Makefile#L15C1-L16C1

Comment (liangyuanpeng) [reply config/jobs/image-pushing/k8s-staging-node-ipam-controller.yaml:26]: Updated, thank you!

**Pull request merged**